src/modules/rtpengine/rtpengine.c

```diff
@@ -3549,6 +3606,10 @@ static int rtpp_test(struct rtpp_node *node, int isdisabled, int force)
 		return 1;
 	}
 	if(force == 0) {
+		/* If ping_interval is set, the timer will ping and test
+		the rtps. No need to do something during routing */
+		if(rtpengine_ping_interval > 0)
+			return 1;
```

Comment:
I think this new if condition needs to be after the existing if(isdisabled==0) condition. Otherwise may interfere with the "found:" label in select_rtpp_node_new() and automatically disable a found node

Reply:
I think you are right, since this function has a bit different semantics that `rtpp_test_ping()`.

This function should return whether the node `isdisabled` or not (not if it's reachable since other checks are in play, like pings or already enabled).

I have made some modification to the check of `ping_interval` and now we return whatever status we already were if ping timer is enabled.

Does this make more sense?

Reply:
I like this change, but I think you should still place it inside "if (force==0) {" condition. Otherwise, when timer pinging is enabled, will cancel all the forced pings of rtpp_test(). More specifically will cancel the effect of (at least) these 2 modparams:
1. aggressive_redetection -> maybe should still be ok since the purpose of the timer pinging is to disable pinging done by the routing processes, but I think should be documented
2. ping_mode -> rtpengines won't be able to be pinged first time when kamailio starts, but will be pinged first time when timer triggers (e.g. after the initial rtpengine_ping_interval seconds). Will all be considered available untill that. Please check the child_init()->build_rtpp_socks()->rtpp_test() flow for this.

Please let me know what you think about this.

Thanks,
Stefan

Reply:
I am currently thinking that if a forced ping is required, regardless of ticks, disabled or not, and so on, the function `rtpp_test_ping` should be used. It does exactly that and only that.

So for 
1. It seems to be ok.
2. We should use the function above instead of `rtpp_test()`

Thoughts on this?

Reply:
I think rtpp_test_ping() and rtpp_test() are very similar, except rtpp_test() skips sending the actual ping in some cases. Definitely first one can replace the second one when force pinging is required (maybe additionally checking manually disabled rtpengines before calling it -> RTPENGINE_MAX_RECHECK_TICKS)

I think, a much more simpler change would be the previous one suggested. So pings will happen additionally, besides the timer pings, only when they are force pings. I wonder if you see any problems with it, that I am missing?

Reply:
Well, i don't really see any problems no. Adding the interval check in `force==0` block can work as expected probably. 

What I suggested was just a thought, that might have simplified how all these are handled and more logically structured.

I will update it accordingly.

Reply:
Updated.

Reply:
Thank you. Looks good to me.